source/slang/slang-ir-legalize-image-subscript.cpp

```diff
@@ -0,0 +1,196 @@
+#include "slang-ir-legalize-image-subscript.h"
+
+#include "slang-ir.h"
+#include "slang-ir-insts.h"
+#include "slang-ir-util.h"
+#include "slang-ir-clone.h"
+#include "slang-ir-specialize-address-space.h"
+#include "slang-parameter-binding.h"
+#include "slang-ir-legalize-varying-params.h"
+
+namespace Slang
+{
+    void legalizeImageSubscriptStore(TargetRequest* target, IRBuilder& builder, IRInst* storeInst, DiagnosticSink* sink)
+    {
+        builder.setInsertBefore(storeInst);
+        auto getElementPtr = as<IRGetElementPtr>(storeInst->getOperand(0));
+        IRImageSubscript* imageSubscript = nullptr;
+        if (getElementPtr)
+            imageSubscript = as<IRImageSubscript>(getElementPtr->getBase());
+        else
+            imageSubscript = as<IRImageSubscript>(storeInst->getOperand(0));
+        assert(imageSubscript);
+        IRTextureType* textureType = as<IRTextureType>(imageSubscript->getImage()->getFullType());
```

Comment:
nullptr check to avoid crash later.

Reply:
I don't exactly know why we need to check for a nullptr here. From what I understand:
1. if `imageSubscript` is `nullptr`, we have a very huge bug in Slang. Earlier in the call stack, the code `if(getRootAddr)` ensures we have a valid `imageSubscript` before hand.
2. if `imageSubscript->getImage()` is `nullptr` the code would have failed already and the language-server would not work with textures.

Reply:
Well, we just don't want to see there is nothing prompted except a "segment fault" if any unexpected bad thing happens.

Reply:
I can add assert's

Reply:
Are these recommended checks for debug and release mode or only debug mode (developers)?

Reply:
Debug check is fine.